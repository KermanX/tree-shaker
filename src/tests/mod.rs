use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
