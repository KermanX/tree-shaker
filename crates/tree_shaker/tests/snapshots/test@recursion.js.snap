---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/recursion.js
---
export function simple() {
	let t = 0;
	function main(x) {
		if (t) {
			sideeffect(2);
		}
		t++;
		return x > 100 ? x : main(2 * x + 1);
	}
	return main(1);
}
export function complex1() {
	function main(a) {
		return () => g(a);
	}
	function g(a) {
		return () => main(a + 1);
	}
	t = main(1);
}
export function complex2() {
	function main() {
		enterHooks = resolveTransitionHooks((hooks) => enterHooks = hooks);
	}
	function resolveTransitionHooks(postClone) {
		const hooks = { clone() {
			const hooks2 = resolveTransitionHooks(postClone);
			if (postClone) postClone(hooks2);
			return hooks2;
		} };
		return hooks;
	}
	return main;
}
export function closure_not_recused() {
	return "a3b";
}
